Source/DafnyServer/boogie/DafnyProvider.cs

```diff
@@ -80,6 +79,29 @@ public DafnyModel(Model m, ViewOptions opts)
       }
     }
 
+    /// <summary>
+    /// Creates a new function that merges together the applications of all the
+    /// functions that have a certain arity and whose name matches the
+    /// "^MapType[0-9]*Select$" pattern. This has previously been done by
+    /// Boogie's Model Parser as a preprocessing step but has been deprecated
+    /// since 2.9.2
+    /// </summary>
+    private Model.Func MergeMapSelectFunctions(int arity) {
+      int id = arity - 1;
+      // Need to make sure the name of the new function is unique
+      while (model.HasFunc("[" + ++id + "]")) { }
```

Comment:
Rewrite this loop to something less cryptic (in particular, something that does not use `++id` in the middle of an expression).

Reply:
I have rewritten the loop and amended the respective comment. Thank you!